tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1025,9 +1029,13 @@ void HWCommandQueue::enqueue_write_buffer(const Buffer& buffer, const void* src,
         dst_page_index += pages_to_write;
     }
 
-    if (buffer.buffer_layout() == TensorMemoryLayout::WIDTH_SHARDED or
-        buffer.buffer_layout() == TensorMemoryLayout::BLOCK_SHARDED) {
+    if (is_sharded(buffer.buffer_layout())) {
+        if ((buffer.buffer_layout() == TensorMemoryLayout::WIDTH_SHARDED or
+            buffer.buffer_layout() == TensorMemoryLayout::BLOCK_SHARDED) or
+            (buffer.buffer_layout() == TensorMemoryLayout::HEIGHT_SHARDED
```

Comment:
Why is this rechecked if it is already inside `if (is_sharded(buffer.buffer_layout()))?`